libs/datasets/dataset_orchestrator.py

```diff
@@ -0,0 +1,318 @@
+            .sort_index()
+            .rename_axis(columns=PdFields.VARIABLE)
+        )
+        static_df = (
+            pd.concat(dataset.static for dataset in datasets)
+            .sort_index()
+            .rename_axis(columns=PdFields.VARIABLE)
+        )
+        tag = pd.concat([dataset.tag for dataset in datasets]).sort_index()
+        multiregion_dataset = MultiRegionDataset(
+            timeseries_bucketed=timeseries_df, static=static_df, tag=tag
+        )
+
+        if generate_hsas:
+            _log.info("Starting HSA aggregation...")
+            hsa_aggregator = statistical_areas.CountyToHSAAggregator.from_local_data()
+            multiregion_dataset = hsa_aggregator.aggregate(multiregion_dataset)
+
+        if generate_cbsas:
+            _log.info("Starting CBSA aggregation...")
+            cbsa_dataset = self.cbsa_aggregator.aggregate(
+                multiregion_dataset, reporting_ratio_required_to_aggregate=DEFAULT_REPORTING_RATIO
+            )
+            multiregion_dataset = multiregion_dataset.append_regions(cbsa_dataset)
+            multiregion_dataset.print_stats("CBSA dataset")
+
+        if aggregate_to_country:
+            _log.info("Starting country aggregation...")
+            multiregion_dataset = custom_aggregations.aggregate_to_country(
+                multiregion_dataset, reporting_ratio_required_to_aggregate=DEFAULT_REPORTING_RATIO
+            )
+            multiregion_dataset.print_stats("Aggregate to country")
+        return multiregion_dataset
+
+    def _build_region_timeseries(self, region_dataset: OneStateDataset):
+        state = region_dataset.state
+        multiregion_dataset = region_dataset.multiregion_dataset
+
+        # TODO: maybe filter overrides to match only the current region?
+        # Not sure if that will help performance but worth a shot.
+        region_overrides_config = manual_filter.transform_region_overrides(
+            self.region_overrides, self.cbsa_aggregator.cbsa_to_counties_region_map
+        )
+        before_manual_filter = multiregion_dataset
+        multiregion_dataset = manual_filter.run(multiregion_dataset, region_overrides_config)
+        manual_filter_touched = manual_filter.touched_subset(
+            before_manual_filter, multiregion_dataset
+        )
+        manual_filter_touched.write_to_wide_dates_csv(
+            dataset_utils.MANUAL_FILTER_REMOVED_WIDE_DATES_CSV_PATH,
+            dataset_utils.MANUAL_FILTER_REMOVED_STATIC_CSV_PATH,
+        )
+        if self.print_stats:
+            multiregion_dataset.print_stats("manual filter")
+        multiregion_dataset = timeseries.drop_observations(
+            multiregion_dataset, after=datetime.datetime.utcnow().date()
+        )
+
+        multiregion_dataset = outlier_detection.drop_tail_positivity_outliers(multiregion_dataset)
+        if self.print_stats:
+            multiregion_dataset.print_stats(f"drop_tail {multiregion_dataset.location_ids}")
```

Comment:
Is this going to be a long list of county FIPS codes or something?  Perhaps just include the state code?